Dockerfile

```diff
@@ -26,9 +26,9 @@ ADD . .
 ARG client_source=k8s
 ENV EFS_CLIENT_SOURCE=$client_source
 
-RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} make aws-efs-csi-driver
+RUN OS=${TARGETOS} ARCH=${TARGETARCH} make $TARGETOS/$TARGETARCH
```

Comment:
curious the purpose of the change on make here?

Reply:
i wanted to make it more consistent with fsx and ebs Makefiles so I copied this pattern from there. Specifically I need the `make image` rule for the e2e test to work beacuse the e2e test needs to build/push an image to a per-test tag and the current rule was pushing it to hardcoded `master` tag which is not usable since multijple tests could be running in parallel.

As for why ebs and fsx are this in the first place, it's because ebs must build for windows so I made a bunch of changes to its Makefile to use these generic OS/ARCH env variables. That comes at cost of readability/simplicity since for efs we only care about arm/x64... so I am open to simplifying the Makefile but probably not in the near future :sweat_smile: